lib/scala/downloader/src/main/scala/org/enso/downloader/archive/TarGzWriter.scala

```diff
@@ -0,0 +1,126 @@
+package org.enso.downloader.archive
+
+import org.apache.commons.compress.archivers.tar.{
+  TarArchiveEntry,
+  TarArchiveOutputStream
+}
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream
+import org.enso.cli.task.{TaskProgress, TaskProgressImplementation}
+
+import java.io.{BufferedOutputStream, FileOutputStream}
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+import scala.util.{Try, Using}
+
+/** A helper class for writing TAR archives compressed with gzip. */
+class TarGzWriter private (archive: TarArchiveOutputStream) {
+
+  /** Adds a text file to the archive.
+    *
+    * @param relativePath path of the file in the archive
+    * @param content the text content to put in the file
+    */
+  def writeTextFile(relativePath: String, content: String): Unit = {
+    val bytes = content.getBytes(StandardCharsets.UTF_8)
+    val entry = new TarArchiveEntry(relativePath)
+    entry.setSize(bytes.size.toLong)
+    archive.putArchiveEntry(entry)
+    archive.write(bytes)
+    archive.closeArchiveEntry()
```

Comment:
What happens if these fail? There doesn't seem to be any failure handling here.

Reply:
They just throw an exception which bubbles up throught the nested calls, it stops writing of the archive and closes it (if anything was written it will be partially written) and bubbles up to the createArchive call.

Reply:
Okay, sounds good.